src/jna-client/org/firebirdsql/gds/ng/jna/AbstractNativeDatabaseFactory.java

```diff
@@ -35,14 +32,14 @@
 public abstract class AbstractNativeDatabaseFactory implements FbDatabaseFactory {
 
     @Override
-    public JnaDatabase connect(IConnectionProperties connectionProperties) throws SQLException {
+    public FbDatabase connect(IConnectionProperties connectionProperties) throws SQLException {
```

Comment:
Why did you remove the covariant inheritance here? This also introduces casts in TestJnaDatabase and TestJnaBlob that were previously unnecessary.

Reply:
I forgot. It was a rough initial version, need to reverse it.